presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -191,17 +201,47 @@ public void startWaitingForResources()
 
     private void waitForMinimumWorkers()
     {
-        ListenableFuture<?> minimumWorkerFuture = clusterSizeMonitor.waitForMinimumWorkers();
+        int nodesToAcquire = computeNodesToAcquire();
+        CompletableFuture<?> waitForNodesFuture =
+                nodeScheduler.acquireNodes(stateMachine.getQueryId(), nodesToAcquire, stateMachine::getQueryState);
```

Comment:
How do you handle cluster turnup now ? We are no longer waiting for the waitForMinWorkers ? See comment/suggestion on handling that in the fullfillPendingRequest

Reply:
Good point. Will definitely look into this past the POC phase. The current POC is a very controlled environment, so didn't fix this.